pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -192,7 +192,17 @@ def restore_training_state(self) -> None:
             return
 
         # restore precision plugin (scaler etc.)
-        self.trainer.precision_plugin.on_load_checkpoint(self._loaded_checkpoint)
+        prec_plugin = self.trainer.precision_plugin
+        prec_plugin.on_load_checkpoint(self._loaded_checkpoint)
+        if prec_plugin.__class__.__qualname__ in self._loaded_checkpoint:
```

Comment:
Here you do `on_load_checkpoint` -> `load_state_dict`

but below: `state_dict` -> `on_save_checkpoint`.

Is this on purpose?

Reply:
I would always have the old api falling back to the new one with a warning and *never* the other way around :)

Reply:
save: `state_dict` needs to come before `on_save_checkpoint` in order for `on_save_checkpoint` to receive the "full" ckpt dict argument.

load: not as definitive as save, but I was thinking `on_load_checkpoint` gets the loaded ckpt dict to do any necessary logic before passing it to `load_state_dict` calls - thoughts?

Reply:
@jjenniferdai please add that as a comment directly in the code